demisto_sdk/commands/update_release_notes/update_rn.py

```diff
@@ -80,55 +110,371 @@ def get_deprecated_comment_from_desc(description: str) -> str:
+    """
+    if new_content.deprecated and not old_content.deprecated:
+        if isinstance(new_content, Argument) and parent:
+            return DEPRECATED_ARGUMENT.format(
+                name=name, type=content_type.value, command_name=parent
+            )
+        return GENERAL_DEPRECATED_RN.format(
+            name=name,
+            type=content_type.value,
+            replacement=(
+                get_deprecated_comment_from_desc(new_content.details_for_rn())
+                or "Use %%% instead"
+            ),
+        )
+    return ""
+
+
+def generate_required_rn(
+    name: str,
+    new_content: Union[Parameter, Argument, Command],
+    old_content: Union[Parameter, Argument, Command],
+    content_type: content_type,
+) -> str:
+    """Generates release notes for content items that have been updated to be required.
+
+    Args:
+        name (str): The name of the content item being updated.
+        new_content (Union[Parameter, Argument, Command]): The updated version of the content item.
+        old_content (Union[Parameter, Argument, Command]): The previous version of the content item.
+        content_type (content_type): The type of the content item (e.g., parameter, argument, command).
+
+    Returns:
+        str: A formatted release note indicating that the specified content item is now required.
+             Returns an empty string if no changes to the "required" status are detected.
+    """
+    if (
+        (
+            (isinstance(new_content, Parameter) and isinstance(old_content, Parameter))
+            or (isinstance(new_content, Argument) and isinstance(old_content, Argument))
+        )
+        and new_content.required
+        and not old_content.required
+    ):
+        return f"- Updated the **{name}** {content_type.value} to be required.\n"
+    return ""
+
+
+def generate_new_content_rn(
+    name: str,
+    new_content: Union[Parameter, Argument, Command],
+    content_type: content_type,
+    parent: Union[str, None] = None,
+) -> str:
+    """Generates release notes for newly added content of content items.
+
+    Args:
+        item_name (str): The name of the new content item.
+        new_content_data (dict[str, Any]): The data for the new content item.
+        content_type (content_type): The type of the content item (e.g., parameter, argument, command).
+        parent_name (str | None, optional): The name of the parent content item, if applicable (e.g., command name for arguments).
+
+    Returns:
+        str: A formatted release note describing the addition of the content item.
+    """
+    description = new_content.details_for_rn().replace("-", "").lower()
+    display_name = new_content.name_for_rn()
+    if isinstance(new_content, Argument) and parent:
+        return ADDED_ARGUMENT_RN.format(name=name, parent=parent)
+    extra_description = f" which {description.rstrip('.')}." if description else "."
```

Comment:
ask shirly if "which" can fit here

Reply:
Not sure what line 278 means.  BTW, I do not review py files.  Only yml and md files.